### DISCUSSION TITLE
How to make a text copy to clipboard

### DISCUSSION THREAD

**Discussion by Nagakirthi** [community/community#138455] [Other Feature Feedback, Questions, & Ideas]
How to make a text copy to clipboard

**Comment by sabbirsami** [comment]
```JS
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Copy to Clipboard</title>
</head>
<body>
    <textarea id="text-to-copy">This is the text to copy!</textarea>
    <button id="copy-button">Copy Text</button>

    <script>
        document.getElementById('copy-button').addEventListener('click', () => {
            const textToCopy = document.getElementById('text-to-copy').value;
            navigator.clipboard.writeText(textToCopy)
                .then(() => {
                    alert('Text copied to clipboard!');
                })
                .catch(err => {
                    console.error('Failed to copy text: ', err);
                });
        });
    </script>
</body>
</html>
```

### **Explanation**


- Clipboard API: This is the modern way to copy text to the clipboard using navigator.clipboard.writeText(). It supports promises, making it easier to handle asynchronous operations.

- document.execCommand: An older method that requires selecting the text first and then calling document.execCommand('copy'). It is widely supported but considered less modern.

_For new projects, the Clipboard API is recommended due to its simplicity and better support for modern web practices._

**Comment by lqduy** [comment]
This function can operate over HTTPS:

```ts
export const copyToClipboard = async (text: string): Promise<boolean> => {
  try {
    if (text) {
      const input = document.createElement("textarea")
      input.value = text
      document.body.appendChild(input)
      input.select()
      document.execCommand("copy")
      document.body.removeChild(input)
      return true
    }
    return false
  } catch (error) {
    return false
  }
}
```

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @Nagakirthi ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!
